src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -21,6 +34,235 @@ private SslStreamCertificateContext(X509Certificate2 target, X509Certificate2[]
             SslContexts = new ConcurrentDictionary<SslProtocols, SafeSslContextHandle>();
         }
 
-        internal static SslStreamCertificateContext Create(X509Certificate2 target) => Create(target, null);
+        internal static SslStreamCertificateContext Create(X509Certificate2 target) =>
+            Create(target, null, offline: false, trust: null, noOcspFetch: true);
+
+        internal bool OcspStaplingAvailable => _ocspUrls is not null;
+
+        partial void SetNoOcspFetch(bool noOcspFetch)
+        {
+            _staplingForbidden = noOcspFetch;
+        }
+
+        partial void AddRootCertificate(X509Certificate2? rootCertificate)
+        {
+            if (IntermediateCertificates.Length == 0)
+            {
+                _ca = rootCertificate;
+            }
+            else
+            {
+                _ca = IntermediateCertificates[0];
+            }
+
+            if (!_staplingForbidden)
+            {
+                // Create the task, let the download finish in the background.
+                GetOcspResponseAsync().AsTask();
+            }
+        }
+
+        internal byte[]? GetOcspResponseNoWaiting()
+        {
+            try
+            {
+                ValueTask<byte[]?> task = GetOcspResponseAsync();
+
+                if (task.IsCompletedSuccessfully)
+                {
+                    return task.Result;
+                }
+            }
+            catch
+            {
+            }
+
+            return null;
+        }
+
+        internal ValueTask<byte[]?> GetOcspResponseAsync()
+        {
+            if (_staplingForbidden)
+            {
+                return ValueTask.FromResult((byte[]?)null);
+            }
+
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+
+            if (now > _ocspExpiration)
+            {
+                return DownloadOcsp();
+            }
+
+            if (now > _nextDownload)
+            {
+                // Calling DownloadOcsp will activate a Task to initiate
+                // in the background.  Further calls will attach to the
+                // same Task if it's still running.
+                //
+                // We don't want the result here, just the task to background.
+#pragma warning disable CA2012 // Use ValueTasks correctly
+                DownloadOcsp();
+#pragma warning restore CA2012 // Use ValueTasks correctly
+            }
+
+            return ValueTask.FromResult(_ocspResponse);
+        }
+
+        private ValueTask<byte[]?> DownloadOcsp()
+        {
+            Task<byte[]?>? pending = _pendingDownload;
+
+            if (pending is not null && !pending.IsFaulted)
```

Comment:
This never needs to transition from non-null back to null?

Reply:
`_pendingDownload` goes back to null if it completes successfully.  On... line.... that got deleted.